### PR TITLE
Replace MathUtils.clamp with Kotlin coerceIn

### PR DESCRIPTION
**Pull request by TacoTheDank**
#### What is it?
- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [x] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR
- Replace 'MathUtils.clamp' with Kotlin 'coerceIn' where possible
- Also utilize the isGone extension function

#### APK testing
The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR. You can find more info and a video demonstration [on this wiki page](https://github.com/TeamNewPipe/NewPipe/wiki/Download-APK-for-PR).

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

File: app/src/main/java/org/schabi/newpipe/player/gesture/PopupPlayerGestureListener.kt [hunk 1]
```diff
@@ -251,8 +249,7 @@ class PopupPlayerGestureListener(
         // -- Determine if the ClosingOverlayView (red X) has to be shown or hidden --
         val showClosingOverlayView: Boolean = playerUi.isInsideClosingRadius(movingEvent)
         // Check if an view is in expected state and if not animate it into the correct state
-        val expectedVisibility = if (showClosingOverlayView) View.VISIBLE else View.GONE
-        if (binding.closingOverlay.visibility != expectedVisibility) {
+        if (binding.closingOverlay.isGone) {
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/player/gesture/PopupPlayerGestureListener.kt:252]: I think this is what you meant
```suggestion
        if (binding.closingOverlay.isVisible != showClosingOverlayView) {
```

**Review by Stypox — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/schabi/newpipe/player/gesture/PopupPlayerGestureListener.kt (hunk 1 above)
Comment (TacoTheDank) [reply app/src/main/java/org/schabi/newpipe/player/gesture/PopupPlayerGestureListener.kt:252]: oh my god

7sj2qs

**Review by AudricV — DISMISSED** [review]
Are you sure about your changes? A part of them has been intentionally reverted with #8737.

**Comment by TacoTheDank** [comment]
@AudricV That's interesting, I wasn't aware of that.

If the problem is that `(playerUi.screenWidth - playerUi.popupLayoutParams.width).toFloat()` or `(playerUi.screenHeight - playerUi.popupLayoutParams.height).toFloat()` can somehow result in a negative number and therefore crash because it is less than the minimum (0f), then I think we can just use Kotlin's `coerceAtLeast` to make sure numbers can't go below '0f'. `coerceAtLeast` looks like this:

```
public fun Float.coerceAtLeast(minimumValue: Float): Float {
    return if (this < minimumValue) minimumValue else this
}
```

This would ensure we always have a maximum of at least '0f' and avoid a crash.

**Review by Stypox — APPROVED** [review]

**Comment by Stypox** [comment]
ktlint fails

**Review by TobiGr — APPROVED** [review]
LGTM

**Pull request merged**